Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2770,25 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-
+  FServerInfo.ServerInfo.Clear;
+  if IsExtSupported('CSID') then begin {do not localize}
+      SendCmd('CSID ' + FClientInfo.GetCSIDOutput);  {do not localize}
+      if Self.LastCmdResult.NumericCode = 200 then begin
+        LBuf := TrimRight(Self.LastCmdResult.Text.Text);
+        repeat
+           FServerInfo.ServerInfo.Add(TrimLeft(Fetch(LBuf,';')));

```

Comment:
Perhaps replace this whole loop with a single call to `IdGlobal.SplitDelimitedString()` instead?

`SplitDelimitedString(LastCmdResult.Text.Text, FServer.ServerInfo, True, ';');`